port/mmap.h

```diff
@@ -67,4 +68,20 @@ class MemMapping {
   static MemMapping AllocateAnonymous(size_t length, bool huge);
 };
 
+template <typename T>
+class TypedMemMapping : public MemMapping {
+ public:
+  /*implicit*/ TypedMemMapping(MemMapping&& v) noexcept
```

Comment:
Should this and below move constructor take `TypedMemMapping` as input?

Reply:
I didn't see it important to create another set of factory functions for this type, so you first construct a MemMapping. Because TypedMemMapping is-a MemMapping with no additional fields, you can use these move operators on TypedMemMapping also. I intended it as a simple convenience, not a bullet-proof full-featured API. ;)